genie-web/src/main/java/com/netflix/genie/web/data/services/PersistenceService.java

```diff
@@ -970,6 +970,18 @@ void updateJobArchiveStatus(
      * @return A {@link JobInfoAggregate} containing the metadata information
      */
     JobInfoAggregate getHostJobInformation(@NotBlank String hostname);
+
+    /**
+     * Get the set of agent jobs that are satisfy the following criteria.
+     * - Agent job
+     * - Final state
+     * - Finished time earlier than N seconds ago
+     * - Archive status is PENDING
+     *
+     * @param updatedThreshold only select jobs update before this threshold
+     * @return The list of job ids
+     */
+    Set<String> getFinishedJobsWithPendingArchiveStatus(@NotNull Instant updatedThreshold);
```

Comment:
Any thought to making this a bit more generic? `getJobsWithStatusAndArchiveStatus` or something? That way it could be used for more use cases in future if necessary without developing entirely new API

Reply:
Done